dids/diddht/diddht.go

```diff
@@ -1,175 +1,191 @@
-		case strings.HasPrefix(name, "_s"):
-			var s didcore.Service
-			if err := UnmarshalService(data, &s); err != nil {
-				// TODO handle error
-				continue
+// WithVerificationMethod adds the provided verification method to the DID document.
+func WithVerificationMethod(method didcore.VerificationMethod, purposes []didcore.Purpose) DHTDidOption {
+	return func(o *dHTDidOptions) {
+		o.verificationMethods = append(o.verificationMethods, method)
+		for _, p := range purposes {
+			if _, ok := o.purposes[method.ID]; !ok {
+				o.purposes[method.ID] = []didcore.Purpose{}
 			}
-			document.AddService(&s)
-		case strings.HasPrefix(name, "_cnt"):
-			// TODO add controller https://did-dht.com/#controller
-			// optional field
-			// comma-separated list of controller DID identifiers
-			document.Controller = strings.Split(data, ",")
-		case strings.HasPrefix(name, "_aka"):
-			// TODO add aka https://did-dht.com/#also-known-as
-			document.AlsoKnownAs = strings.Split(data, ",")
-		default:
+			o.purposes[method.ID] = append(o.purposes[method.ID], p)
 		}
 	}
+}
 
-	return document, nil
+// WithKeyManager sets the key manager to use for generating the DID's keypair.
+func WithKeyManager(k crypto.KeyManager) DHTDidOption {
+	return func(o *dHTDidOptions) {
+		o.keyManager = k
+	}
 }
 
-// parseDNSDID takes the bytes of the DNS representation of a DID and creates an internal representation
-// used to create a DID document
-func parseDNSDID(data []byte) (*Decoder, error) {
-	var p dnsmessage.Parser
-	if _, err := p.Start(data); err != nil {
-		return nil, err
+// WithRelay sets the relay to use for publishing the DID to the DHT.
+func WithRelay(relayURL string, client *http.Client) DHTDidOption {
+	return func(o *dHTDidOptions) {
+		o.relay = pkarr.NewPkarrRelay(relayURL, client)
 	}
+}
+
+// Create creates a new `did:dht` DID and publishes it to the DHT network via a Pkarr relay.
+//
+// If no relay is passed in the options, Create uses a default Pkarr relay.
+// Spec: https://did-dht.com/#create
+func Create(opts ...DHTDidOption) (*did.BearerDID, error) {
+	return CreateWithContext(context.Background(), opts...)
+}
 
-	didRecord := Decoder{
-		records: make(map[string]string),
+func CreateWithContext(ctx context.Context, opts ...DHTDidOption) (*did.BearerDID, error) {
+
+	// 0. Set default options
+	o := dHTDidOptions{
+		algorithmID:         dsa.AlgorithmIDED25519,
+		verificationMethods: []didcore.VerificationMethod{},
+		services:            []*didcore.Service{},
+		keyManager:          crypto.NewLocalKeyManager(),
+		relay:               getDefaultRelay(),
+		purposes:            map[string][]didcore.Purpose{},
 	}
 
-	// need to skip questions to move the index to the right place to read answers
-	if err := p.SkipAllQuestions(); err != nil {
-		return nil, err
+	for _, opt := range opts {
+		opt(&o)
 	}
 
-	for {
-		h, err := p.AnswerHeader()
-		if errors.Is(err, dnsmessage.ErrSectionDone) {
-			break
-		}
+	// 1. Generate an Ed25519 keypair
```

Comment:
well it's not guaranteed to be `Ed25519` right? could be caller-overridden

Reply:
```suggestion
	// 1. Generate keypair
```

Reply:
I see your point, I thought the same thing, but it's literally the 1st step in the spec, and it's specific https://did-dht.com/#create